### PR TITLE
Handle unique constraint violation error when signup

### PR DESCRIPTION
**Pull request by Steven-Chan**
connect #393

### CODE REVIEW

File: pkg/server/skydb/pq/recorddb.go [hunk 1]
```diff
@@ -150,6 +150,15 @@ func (db *database) Save(record *skydb.Record) error {
 
 	row := db.c.QueryRowWith(upsert)
 	if err = newRecordScanner(record.ID.Type, typemap, row).Scan(record); err != nil {
+		if pqErr, ok := err.(*pq.Error); ok {
```
Comment (cheungpat) [inline pkg/server/skydb/pq/recorddb.go]: `if isUniqueViolated {`

**Comment by cheungpat** [comment]
generally okay but there are test errors that need fixing

**Comment by Steven-Chan** [comment]
skydb/skydbtest/skydbmockutil.go
```
func ExpectDBSaveUser(db *mock_skydb.MockTxDatabase, extendedSchema *skydb.RecordSchema, assertSavedUserRecord interface{}, saveError error) {
	db.EXPECT().ID().Return("_public").AnyTimes()

	// no record found
	db.EXPECT().
		Get(gomock.Any(), gomock.Any()).
		Return(skydb.ErrRecordNotFound).
		AnyTimes()

        // db.Get(skydb.NewRecordID("user", "abc"), nil) === skydb.ErrRecordNotFound
        // it works pretty well

	// extend Schema
	if extendedSchema != nil {
		ExpectDBExtendSchema(db, *extendedSchema)
	}

	db.EXPECT().
		Save(gomock.Any()).
		Do(assertSavedUserRecord).
		Return(saveError).
		AnyTimes()

         // db.Save(nil) == nil
         // saveError != nil
         // WHAT THE HELL IS GOING ON
}
```

@cheungpat 
Because of this ridiculous behaviour of gomock, I am going to delete those test case for auth data duplication.

**Comment by Steven-Chan** [comment]
@cheungpat updated

**Pull request merged**